packages/jest-leak-detector/src/index.ts

```diff
@@ -46,12 +49,15 @@ export default class {
     value = null;
   }
 
-  isLeaking(): Promise<boolean> {
+  async isLeaking(): Promise<boolean> {
     this._runGarbageCollector();
 
-    return new Promise(resolve =>
-      setImmediate(() => resolve(this._isReferenceBeingHeld)),
-    );
+    // wait some ticks to allow GC to run properly, see https://github.com/nodejs/node/issues/34636#issuecomment-669366235
+    for (let i = 0; i < 10; i++) {
```

Comment:
measured, it takes about 0.3ms on my machine when running this module's unit tests